### PR TITLE
Josi McClellan - Open Mic

### PR DESCRIPTION
**Pull request by JosiahMcClellan**

### CODE REVIEW

File: lib/joke.rb [hunk 1]
```diff
@@ -1,3 +1,10 @@
 class Joke
 
+  attr_reader :id, :question, :answer
+  def initialize(id:, question:, answer:)
```
Comment (VictoriaVasys) [inline lib/joke.rb]: It's probably a better idea to accept a single param for your joke hash (e.g. `joke_info`); you would run into problems if someone passed joke info with additional key-value pairs. Look more into best use cases for named parameters.

File: test/user_test.rb [hunk 2]
```diff
@@ -21,4 +22,94 @@ def test_it_can_have_another_name
...
+    assert_instance_of Array, User.new("Lapis").jokes
+  end
+
+  def test_it_starts_with_no_jokes
+    assert User.new("Lapis").jokes.empty?
+  end
+
+  def test_it_can_learn_jokes
+    user = User.new("Lapis")
+    joke = strawberry_joke
```
Comment (VictoriaVasys) [inline test/user_test.rb]: This seems extraneous; using strawberry_joke isn't going to make your lines too long & would result in less round-about setup

File: test/user_test.rb [hunk 3]
```diff
@@ -21,4 +22,94 @@ def test_it_can_have_another_name
...
+    lapis = User.new("Lapis")
+    lazuli = User.new("Lazuli")
+
+    strawberry = strawberry_joke
+    lion = lion_joke
+    silly = silly_joke
+
+    lapis.learn(strawberry)
+    lapis.learn(lion)
+
```
Comment (VictoriaVasys) [inline test/user_test.rb]: No need for extra spacing here; I would say your tests should have at most 3 'sections' (spaces between chunks of code); one for setup, one for 'actions' and one for assertions

File: lib/joke.rb (hunk 1 above)
Comment (JosiahMcClellan) [reply lib/joke.rb]: Funny, I somehow had it in my head that "named parameters" was just a special syntax for a using a hash as the last argument, not a feature in its own right.  Discovered I was wrong at iteration 5 with a weird error when I tried to pass in `row.to_hash` from the CSV.  Spent all that looking for how to turn it into a hash and it didn't even work.

**Pull request closed without merging**